openmc/mgxs_library.py

```diff
@@ -884,8 +998,10 @@ def set_scatter_mgxs(self, scatter, nuclide='total', xs_type='macro',
                     [self.energy_groups])
         check_value('domain_type', scatter.domain_type,
                     ['universe', 'cell', 'material', 'mesh'])
+        check_type('temperature', temperature, Real)
+        check_value('temperature', temperature, self.temperatures)
 
-        if self.scatt_type != 'legendre':
+        if (self.scatter_format != 'legendre'):
```

Comment:
Also it seems like you use a mixture of `==` vs. `is` and `!=` and vs. `is not` for string comparisons throughout the code. Any reason not to be consistent?